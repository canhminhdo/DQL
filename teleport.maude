load BDQL.maude

fmod TELEPORT is
    pr BDQL-SEMANTICS .

    var S : Scalar .

    --- defining initial states
    ops a b : -> Scalar .
    eq Pow(Abs(a), 2) .+ Pow(Abs(b), 2) = 1 .
    eq S .* Pow(Abs(a), 2) .+ S .* Pow(Abs(b), 2) = S .

    op initQState : -> Qubit .
    eq initQState = (q[0]: a . |0> + b . |1>) (q[1]: |0>) (q[2]: |0>) .
endfm

in debug.maude

red in TELEPORT : initQState |= [ H(1) ; CX(1, 2) ; CX(0, 1) ; H(0) ;
if P(1, |0>) then skip else X(2) fi ;
if P(0, |0>) then skip else Z(2) fi
] P(2, qubitAt(initQState, 0)) .
