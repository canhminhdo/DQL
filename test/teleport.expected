==========================================
search in TELEPORT-CHECK : initQState |= [H(1) ; CX(1,2) ; CX(0,1) ; H(0) ; if
    P(1,|0>) -> skip | P(1,|1>) -> X(2) fi ; if P(0,|0>) -> skip | P(0,|1>) ->
    Z(2) fi]prop =>! JS:JudgementSet such that JS:JudgementSet =/= emptyJS =
    true .
Prob = 1/2
Prob = 1/2
Prob = 1/2
Q = (q[0]: |1>) (q[1]: |0>) q[2]: a . |0> + b . |1>
Prob = 1/2
Q = (q[0]: |0>) (q[1]: |0>) q[2]: a . |0> + b . |1>
Prob = 1/2
Q = (q[0]: |1>) (q[1]: |1>) q[2]: a . |0> + b . |1>
Prob = 1/2
Q = (q[0]: |0>) (q[1]: |1>) q[2]: a . |0> + b . |1>
Prob = 1/2
Q = (q[0]: |1>) (q[1]: |1>) q[2]: a . |0> + b . |1>
Prob = 1/2
Q = (q[0]: |0>) (q[1]: |1>) q[2]: a . |0> + b . |1>
Prob = 1/2
Q = (q[0]: |1>) (q[1]: |0>) q[2]: a . |0> + b . |1>
Prob = 1/2
Q = (q[0]: |0>) (q[1]: |0>) q[2]: a . |0> + b . |1>

No solution.
Bye.
